spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/VectorizedSparkParquetReaders.java

```diff
@@ -75,6 +108,57 @@ public static ColumnarBatchReader buildReader(
                 deleteFilter));
   }
 
+  public static ColumnarBatchReader buildReader(
```

Comment:
This method is added to make sure `NullCheckingForGet.NULL_CHECKING_ENABLED` is referenced after we set system properties in this class. Otherwise, Arrow would memorize them earlier and our logic would have no effect. See `BoundsChecking` and `NullCheckingForGet` in Arrow for details.

Reply:
We never had specific Iceberg configs for this behavior. We always relied on system properties for Arrow.

Reply:
LGTM. 

In our internal Spark fork, we have the arrow config settings configured in the executor JVM args. I can confirm that things have been running fine with them in our prod env for the last 2+ years. So, +1.

Reply:
That's nice to hear, @samarthjain! All of our tests also set these properties so it seems fairly safe.